app/harvesters/nara_harvester.rb

```diff
@@ -0,0 +1,120 @@
+require 'pry'
+
+##
+# A harvester for NARA's API
+#
+# @see Krikri::Harvesters::ApiHarvester
+class NaraHarvester < Krikri::Harvesters::ApiHarvester
+  ##
+  # Initialize, and set default options as appropriate.
+  #
+  # @param opts [Hash] a hash of options as defined by {.expected_opts}
+  #
+  # @example
+  #    Typical instantiation, good for most cases:
+  #        NaraHarvester.new
+  #    Specifying custom parameters:
+  #        NaraHarvester.new(api: {'some_query_param' => 'abc'}, batchsize: 15)
+  #
+  # Parameters to 'opts':
+  # - uri:        See Krikri::Harvester#initialize.
+  #               Defaults to "https://catalog.archives.gov/api/v1"
+  # - name:       See Krikri::Harvester#initialize.  Defaults to "nara"
+  # - batchsize:  The number of records to fetch with each API request.
+  #               Defaults to 10.
+  # - id_fname:   The file name of the Heidrun::IDSource that drives the
+  #               harvest and governs which records to fetch.  Defaults to
+  #               /var/tmp/nara_ids.
+  # - id_fh:      A filehandle to use with Heidrun::IDSource, mostly useful for
+  #               automated testing or console usage.  Optional.
+  #
+  # For other parameters, see Krikri::Harvester#initialize.  
+  #
+  # Parameters for API requests can be specified with the :api key, but have
+  # default values and this should not usually be necessary. See
+  # https://github.com/usnationalarchives/Catalog-API/blob/master/search_and_export.md
+  #
+  def initialize(opts = {})
+    opts[:uri] ||= 'https://catalog.archives.gov/api/v1'
+    opts[:name] ||= 'nara'
+    batchsize = opts.delete(:batchsize) { 10 }
+    # TODO:
+    # @id_source is an enumerator over NARA identifiers (naId values).
+    # This reads from a file of valid IDs in order to work around the fact that
+    # we can not page through NARA's entire result set, due to limitations on
+    # the maximum "offset" value in their API.  When they remove this
+    # limitation from their API, remove @id_source and refactor this method,
+    # #enumerate_records, and #get_count.
+    id_fname = opts.delete(:id_source_filename) { '/var/tmp/nara_ids' }
+    id_fh = opts.delete(:id_source_fh) { File.open(id_fname, 'rt') }
```

Comment:
I'll just have it document with `@raise` that it will raise the error, because the harvester is useless if it can't open that file, whichever it is, the default one or the one that's supplied.